test/jdk/java/awt/Scrollbar/AquaLFScrollbarTest/ScrollBarBorderTest.java

```diff
@@ -0,0 +1,190 @@
+            System.out.println(laf.getName());
+        } catch (UnsupportedLookAndFeelException ignored){
+            System.out.println("Unsupported LookAndFeel: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException |
+                IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void testHorizontal() {
+        // create scroll bar
+        scrollBar = new JScrollBar(JScrollBar.HORIZONTAL);
+        scrollBar.setSize(WIDTH, HEIGHT);
+        scrollBar.setBorder(new HorizontalCustomBorder());
+
+        // paint image with thumb set to default value
+        BufferedImage image1 = new BufferedImage(WIDTH, HEIGHT, TYPE_INT_ARGB);
+        Graphics2D graphics2D = image1.createGraphics();
+        scrollBar.paint(graphics2D);
+        graphics2D.dispose();
+
+        // paint image with thumb set to max value
+        scrollBar.setValue(Integer.MAX_VALUE);
+        BufferedImage image2 = new BufferedImage(WIDTH, HEIGHT, TYPE_INT_ARGB);
+        Graphics2D graphics2D2 = image2.createGraphics();
+        scrollBar.paint(graphics2D2);
+        graphics2D2.dispose();
+
+        // check border for thumb
+        for (int i = WIDTH - BORDER_WIDTH; i < WIDTH; i++) {
+            for (int j = 0; j < HEIGHT; j++) {
+                int c1 = image1.getRGB(i,j);
+                int c2 = image2.getRGB(i,j);
+                if(c1 != c2) {
+                    System.out.println(i + " " + j + " " + "Color1: "
+                                       + Integer.toHexString(c1));
+                    System.out.println(i + " " + j + " " + "Color2: "
+                                       + Integer.toHexString(c2));
+                    saveImage(image1, "himage1.png");
+                    saveImage(image2, "himage2.png");
+                    throw new RuntimeException("Horizontal border has a thumb in it");
+                }
+            }
+        }
+    }
+
+    public static void testVertical() {
+        // create scroll bar
+        scrollBar = new JScrollBar(JScrollBar.VERTICAL);
+        scrollBar.setSize(HEIGHT, WIDTH);
+        scrollBar.setBorder(new VerticalCustomBorder());
+
+        // paint image with thumb set to 0
+        scrollBar.setValue(0);
+        BufferedImage image1 = new BufferedImage(HEIGHT, WIDTH, TYPE_INT_ARGB);
+        Graphics2D graphics2D = image1.createGraphics();
+        scrollBar.paint(graphics2D);
+        graphics2D.dispose();
+
+        // paint image with thumb set to max value
+        scrollBar.setValue(Integer.MAX_VALUE);
+        BufferedImage image2 = new BufferedImage(HEIGHT, WIDTH, TYPE_INT_ARGB);
+        Graphics2D graphics2D2 = image2.createGraphics();
+        scrollBar.paint(graphics2D2);
+        graphics2D2.dispose();
+
+        // check border for thumb
+        for (int i = WIDTH - BORDER_WIDTH; i < WIDTH; i++) {
+            for (int j = 0; j < HEIGHT; j++) {
```

Comment:
Using `x` and `y` would make the loops clearer:
```suggestion
        for (int y = WIDTH - BORDER_WIDTH; y < WIDTH; y++) {
            for (int x = 0; x < HEIGHT; x++) {
```